app/transformations/MemberPaymentsTransformer.scala

```diff
@@ -115,43 +116,85 @@ class MemberPaymentsTransformer @Inject()(
+              s"PSR is Submitted - Member $index state is Changed but has a member PSR version and no previous version - existing member, leaving status as Changed"
+            )
             (index, currentMemberDetail)
           case (((index, currentMemberDetail), _), _)
               if currentMemberDetail.state.changed && initialMemberDetails.get(index).exists(_.state.changed) =>
+            logger.info(
+              s"Member $index state is Changed but initial UA shows the same member as Changed - leaving status as Changed"
+            )
+            (index, currentMemberDetail)
+          case (((index, currentMemberDetail), _), Some(previousMemberDetails))
+              if currentMemberDetail.state.changed && previousMemberDetails.contains(index) =>
+            logger.info(
+              s"Member $index state is Changed and previous version member details shows the same member exists - leaving status as Changed"
+            )
             (index, currentMemberDetail)
-          case (((index, currentMemberDetail), _), Some(previousUA))
-              if currentMemberDetail.state.changed && previousUA.get(MemberStatus(srn, index)).nonEmpty =>
+          case (((index, currentMemberDetail), Some(Compiled)), Some(previousMemberDetails))
+              if currentMemberDetail.state.changed && previousMemberDetails.contains(index) =>
+            logger.info(
+              s"PSR is Compiled - Member $index state is Changed and previous version member details shows the same member exists - leaving status as Changed"
+            )
             (index, currentMemberDetail)
-          case (((index, currentMemberDetail), _), _) =>
+          case (((index, currentMemberDetail), state), previousMemberDetails) =>
+            logger.info(
+              s"Member $index state has not matched any of the previous statements. member state (${currentMemberDetail.state}), PSR state ($state), previous version member details exists (${previousMemberDetails.nonEmpty}) - setting state to New"
+            )
             (index, currentMemberDetail.copy(state = MemberState.New))
         }
       }
 
       // Omit memberPSRVersion if member has changed
+      // Check for empty member payment sections before comparing members
+      // (this is because we send empty records in certain sections for members)
       val memberDetailsWithCorrectVersion: List[MemberDetails] = memberDetailsWithCorrectState.map {
         case (index, currentMemberDetail) =>
           val optInitialMemberDetail = initialMemberDetails.get(index)
+          val refinedCurrentMember =
+            currentMemberDetail
+              .copy(
+                memberLumpSumReceived =
+                  if (currentMemberDetail.memberLumpSumReceived.exists(_.empty)) None
+                  else currentMemberDetail.memberLumpSumReceived,
+                pensionAmountReceived =
+                  if (currentMemberDetail.pensionAmountReceived.contains(0.0)) None
```

Comment:
Above logic checks `.exists(_.empty)`. This one (and the one below) checks `.contains(0.0)`. Should these be the same?

Reply:
.empty is just a function I added to the MemberLumpSumReceived case class, it does the same thing (checks both double values in the case class are 0.0)